elyra/pipeline/handlers.py

```diff
@@ -57,14 +57,15 @@ def get_runtime_processor_type(runtime_type: str, log: Logger, request_path: str
     """
     processor_manager = PipelineProcessorManager.instance()
     if processor_manager.is_supported_runtime(runtime_type):
-        # The endpoint path contains the shorthand version of a runtime (e.g., 'kfp',
-        # 'airflow'). This case and its associated functions should eventually be removed
-        # in favor of using the RuntimeProcessorType name in the request path.
-        log.warning(
-            f"Deprecation warning: when calling endpoint '{request_path}' "
-            f"use runtime type name (e.g. 'KUBEFLOW_PIPELINES', 'APACHE_AIRFLOW') "
-            f"instead of shorthand name (e.g., 'kfp', 'airflow')"
-        )
+        if runtime_type != "local":
+            # The endpoint path contains the shorthand version of a runtime (e.g., 'kfp',
+            # 'airflow'). This case and its associated functions should eventually be removed
+            # in favor of using the RuntimeProcessorType name in the request path.
+            log.warning(
+                f"Deprecation warning: when calling endpoint '{request_path}' "
+                f"use runtime type name (e.g. 'KUBEFLOW_PIPELINES', 'APACHE_AIRFLOW') "
+                f"instead of shorthand name (e.g., 'kfp', 'airflow')"
+            )
         return processor_manager.get_runtime_type(runtime_type)
     elif processor_manager.is_supported_runtime_type(runtime_type):
```

Comment:
If we switch the order of these if statements, checking for runtime-type first, and loosen the case-sensitivity in https://github.com/elyra-ai/elyra/blob/ef4cba3892999028ed616b24fd797fcc5d57c4ca/elyra/pipeline/processor.py#L90 and
https://github.com/elyra-ai/elyra/blob/ef4cba3892999028ed616b24fd797fcc5d57c4ca/elyra/pipeline/runtime_type.py#L48
(upcasing each item for the comparison and lookup) then I think this extra condition could be avoided and similar scenarios from the CLI (that don't go through the handlers) could also be addressed.

Reply:
This is how I originally handled it, but I was worried I was over-complicating 😄 but it is the way I like slightly better since it would allow removal of the hard-coded `local`

Reply:
Trust your instincts! :smile:  (Of course, watch something subtle about all of this bite us!)